Feature/dossier/mc dossier

RyaRom opened the pull request.

RyaRom committed:
add mail sender

RyaRom committed:
add kafka listener

RyaRom committed:
add kafka consumer + refactoring

RyaRom committed:
add: connection with Deal and Dossier

RyaRom committed:
add: new DTOs to dossier + new API endpoint to deal

RyaRom committed:
add: email sender to dossier

RyaRom committed:
add: credit contract creation to dossier

RyaRom committed:
fix: null Inn employment error in deal db

RyaRom committed:
fix: null Inn employment error in deal db

RyaRom committed:
add: new tests for dossier

Anon12421 commented on application/src/main/java/com/application/model/dto/LoanApplicationRequestDTO.java:
```diff
@@ -8,6 +8,7 @@
 import lombok.Builder;
 import lombok.Data;
 
```
test

MichaelDeyev commented on deal/.idea/vcs.xml:
папку .idea нужно добавить в gitignore

MichaelDeyev commented on deal/src/main/java/com/deal/kafka/EmailProducer.java:
```diff
@@ -0,0 +1,27 @@
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
```
@RequestArgsConstructor более подходит для внедрения зависимостей

MichaelDeyev commented on deal/src/main/java/com/deal/model/mapping/DataMapper.java:
```diff
@@ -16,7 +18,8 @@
 import org.mapstruct.Mappings;
 
 @Mapper(componentModel = "spring")
```
На всякий случай предупрежу- в реальной коммерческой разработке практически никогда не используется mapstruct. С ним может быть больше проблем, чем пользы, особенно со сложными сущностями.
Сейчас оставь, но на будущее имей ввиду

MichaelDeyev commented on deal/src/main/java/com/deal/service/ApplicationService.java:
```diff
@@ -16,31 +19,49 @@
 @Service
 @RequiredArgsConstructor
 @Slf4j
+@Transactional
```
избегай @Transactional на классе. Лучше на каждый метод, который сохраняет сущность в БД повешай эту аннотацию, чем на весь класс сразу

MichaelDeyev commented on deal/src/main/java/com/deal/service/ApplicationService.java:
```diff
@@ -125,15 +160,73 @@ public void applicationScoring(FinishRegistrationRequestDTO finishRegistrationRe
+        updateApplicationStatus(application, ApplicationStatus.CREDIT_ISSUED);
+        application.setSignDate(LocalDateTime.now());
+
+        applicationRepo.save(application);
```
тут и 6 строками выше идут одинаковые действия. Подумаю, можно ли соптимизировать и смержить в 1 сохранение

MichaelDeyev commented on dossier/src/main/java/com/dossier/exception/GlobalExceptionHandler.java:
```diff
@@ -0,0 +1,53 @@
+        log.error("Failed to send mail: {}", exception.getMessage());
+        return ErrorMessage.builder()
+                .statusCode(HttpStatus.INTERNAL_SERVER_ERROR.value())
+                .timestamp(new Date())
```
Лучше дату не через new Date(), а используя LocalDateTime.now()

MichaelDeyev commented on dossier/src/main/java/com/dossier/model/dto/CreditDTO.java:
```diff
@@ -0,0 +1,38 @@
+@Data
+@Builder
+@Schema(title = "Credit", description = "Credit's info.")
+public class CreditDTO {
```
А сваггер работает с java records? Если да, то лучше на record поменяй

RyaRom committed:
add gitignore

RyaRom committed:
refactor: better code style

Change ErrorMessage date type to LocalDateTime, delete redundant repository save in ApplicationService

RyaRom committed:
refactor: replace @Data to record in DTO classes

RyaRom committed:
refactor: replace @Data to record and Double to BigDecimal

replace Double to BigDecimal in conveyor module

RyaRom committed:
refactor(deal module): replace Data to record

RyaRom committed:
init: gateway module

RyaRom committed:
add: Gateway module

Add simple gateway module for request routing

RyaRom committed:
refactor: remove useless class files from gateway

RyaRom committed:
fix: null ses code error

RyaRom committed:
refactor: delete IDE settings

MichaelDeyev commented on gateway/mvnw:
```diff
@@ -0,0 +1,259 @@
+#!/bin/sh
```
лучше этот файл в гитигнор добавить

MichaelDeyev commented on gateway/mvnw.cmd:
```diff
@@ -0,0 +1,149 @@
+<# : batch portion
```
лучше этот файл в гитигнор добавить

RyaRom committed:
add mvn files to gitignore

RyaRom committed:
delete mvn wrapper files
